--- a/libtransmission/upnp.c
+++ b/libtransmission/upnp.c
@@ -10,4 +10,5 @@
 #include <assert.h>
 #include <errno.h>
+#include <stddef.h>
 
 #ifdef SYSTEM_MINIUPNP
@@ -221,4 +222,7 @@
         errno = 0;
-        if (UPNP_GetValidIGD (devlist, &handle->urls, &handle->data,
-                             handle->lanaddr, sizeof (handle->lanaddr)) == UPNP_IGD_VALID_CONNECTED)
+        if (UPNP_GetValidIGD (devlist, &handle->urls, &handle->data, handle->lanaddr, sizeof (handle->lanaddr)
+#if MINIUPNPC_API_VERSION >= 18
+                             , NULL, 0
+#endif
+                             ) == UPNP_IGD_VALID_CONNECTED)
         {
